File: Java/src/Act3/Mago.java
package Act3;

public class Mago extends Personaje {
    private Hechizo hechizo;

    public Mago(String nombre, int puntosVida, int puntosAtaque, Hechizo hechizo) {
        super(nombre, puntosVida, puntosAtaque);
        this.hechizo = hechizo;
    }

    public void lanzarHechizo(Personaje enemigo) {
        if (hechizo == null) {
            System.out.println(nombre + " no tiene un hechizo preparado.");
        } else {
            System.out.println(nombre + " lanza el hechizo " + hechizo.getNombre() + " a " + enemigo.getNombre());
            enemigo.recibirDanio(hechizo.getPoder());
        }
    }

    @Override
    public void atacar(Personaje enemigo) {
        System.out.println(nombre + " ataca con su bastón a " + enemigo.getNombre());
        enemigo.recibirDanio(puntosAtaque);
    }
}

File: KotlinCode/src/Act3/Mago.kt
package Act3

class Mago(nombre: String?, puntosVida: Int, puntosAtaque: Int, var hechizo: Hechizo) :
    Personaje(nombre, puntosVida, puntosAtaque) {

        fun lanzarHechizo(enemigo: Personaje) {
        if (hechizo == null) {
            println(nombre + " no tiene un hechizo preparado.")
        } else {
            println(nombre + " lanza el hechizo " + hechizo.nombre + " a " + enemigo.nombre)
            enemigo.recibirDanio(hechizo.poder)
        }
    }

    override fun atacar(enemigo: Personaje) {
        println(nombre + " ataca con su bastón a " + enemigo.nombre)
        enemigo.recibirDanio(puntosAtaque)
    }
}
